microsoft_teams/icon_microsoft_teams/triggers/new_message_received/trigger.py

```diff
@@ -105,10 +105,22 @@ def get_sorted_messages(self, messages_endpoint: str) -> list:
         messages_result = requests.get(messages_endpoint, headers=headers)
         try:
             messages_result.raise_for_status()
+
+
+        # The beta API bombs out every once in a while with Auth denied. Try forcing a refersh of our auth token and
+        # retry getting messages before raising an exception.
         except Exception as e:
-            raise PluginException(cause=f"Could not get messages from Microsoft Graph API."
-                                        f"Get messages result code: {messages_result.status_code}",
-                                  assistance=messages_result.text) from e
+            self.logger.info("Get messages failed, refreshing token and trying again.")
+            time.sleep(10) # sleep for 10 seconds to make sure we're not killing the API
+            headers = self.connection.get_headers(True) # This will force a refresh of our auth token
+            messages_result = requests.get(messages_endpoint, headers=headers)
+            try:
+                messages_result.raise_for_status()
+            except Exception as e:
+                raise PluginException(cause=f"Could not get messages from Microsoft Graph API."
+                                            f"Get messages result code: {messages_result.status_code}",
+                                      assistance=messages_result.text) from e
+
```

Comment:
I think that we should invest the time into adding additional error handling here so that:

1. Users receive more specific messaging regarding what error is received
2. We don't retry when the request if it's not necessary

Reply:
Unfortunately, Microsoft only returns a 400 - Unauthorized here. The response.text will be a json object with that as one of the text fields, along with other stuff. 

That 400 can be for a variety of reasons, bad API token, bad Refresh token, bad creds, team id changed, etc...

I'd rather generically try to catch everything there, try one reauth attempt, and bail if I can't get messages on the 2nd try. 

If we make it this far in the code and haven't hit an error yet, I have no clue what could be wrong honestly, and the beta error messages aren't very helpful.

Reply:
That's fair. Something for us to watch for in the future. Hopefully we'll get some better messaging as the API matures.